hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/RecordStreamSigningUtils.java

```diff
@@ -0,0 +1,287 @@
+import static com.swirlds.platform.util.FileSigningUtils.signData;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.UnsafeByteOperations;
+import com.hedera.node.app.hapi.utils.exports.FileCompressionUtils;
+import com.hedera.services.stream.proto.HashAlgorithm;
+import com.hedera.services.stream.proto.HashObject;
+import com.hedera.services.stream.proto.RecordStreamFile;
+import com.hedera.services.stream.proto.SignatureFile;
+import com.hedera.services.stream.proto.SignatureObject;
+import com.swirlds.common.crypto.DigestType;
+import com.swirlds.common.crypto.HashingOutputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SignatureException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * Utility class for signing record stream files
+ */
+public class RecordStreamSigningUtils {
+    /**
+     * Hidden constructor
+     */
+    private RecordStreamSigningUtils() {}
+
+    /**
+     * Supported stream version file
+     */
+    public static final int SUPPORTED_STREAM_FILE_VERSION = 6;
+
+    /**
+     * a messageDigest object for digesting entire stream file and generating entire record stream
+     * file hash
+     */
+    private static MessageDigest streamDigest;
+
+    /**
+     * a messageDigest object for digesting metaData in the stream file and generating metaData
+     * hash. Metadata contains: record stream version || HAPI proto version || startRunningHash ||
+     * endRunningHash || blockNumber, where || denotes concatenation
+     */
+    private static MessageDigest metadataStreamDigest;
+
+    /**
+     * Generates a signature file for the given record stream file
+     *
+     * @param signatureFileDestination the full path where the signature file will be generated
+     * @param streamFileToSign         the stream file to be signed
+     * @param keyPair                  the keyPair used for signing
+     * @param hapiVersion              the hapi protobuf version
+     * @return true if the signature file was generated successfully, false otherwise
+     */
+    public static boolean signRecordStreamFile(
+            @NonNull final Path signatureFileDestination,
+            @NonNull final Path streamFileToSign,
+            @NonNull final KeyPair keyPair,
+            @NonNull final String hapiVersion) {
+
+        Objects.requireNonNull(signatureFileDestination, "signatureFileDestination must not be null");
```

Comment:
IMO there's no need for the " must not be null"s - it's throwing an NPE.  That's informative enough.

Reply:
It's just a message when null pointer thrown.

Reply:
yes and the message starts with "NullPointerException: ......" when you finally see it at the console/in a log.  That's all I meant.

Reply:
Assumed everyone knows NullPointerException.